numpy/core/tests/test_numerictypes.py

```diff
@@ -446,6 +446,36 @@ def test_other(self, t):
         assert_equal(np.maximum_sctype(t), t)
 
 
+class Test_sctype2char(object):
+    # This function is old enough that we're really just documenting the quirks
+    # at this point.
+
+    def test_scalar_type(self):
+        assert_equal(np.sctype2char(np.double), 'd')
+        assert_equal(np.sctype2char(np.int_), 'l')
+        assert_equal(np.sctype2char(np.unicode_), 'U')
+        assert_equal(np.sctype2char(np.bytes_), 'S')
+
+    def test_other_type(self):
+        assert_equal(np.sctype2char(float), 'd')
+        assert_equal(np.sctype2char(list), 'O')
+        assert_equal(np.sctype2char(np.ndarray), 'O')
+
+    def test_third_party_scalar_type(self):
+        from numpy.core._rational_tests import rational
+        assert_raises(KeyError, np.sctype2char, rational)
+        assert_raises(KeyError, np.sctype2char, rational(1))
```

Comment:
These results are dumb, but IMO the existence of this function is also dumb. Using `np.dtype(rational).char` would give `'r'` here, which is more sensible - but ultimately I can't think of a reason you'd want the char in the first place.